tagstudio/src/qt/widgets/video_player.py

```diff
@@ -0,0 +1,205 @@
+		self.play_pause.setMouseTracking(True)
+		self.play_pause.installEventFilter(self)
+		self.scene().addWidget(self.play_pause)
+		self.play_pause.resize(100, 100)
+		self.play_pause.move(self.width()/2 - self.play_pause.size().width()/2, self.height()/2 - self.play_pause.size().height()/2)
+		self.play_pause.hide()
+
+		self.mute_button = QSvgWidget('./tagstudio/resources/volume_muted.svg', self)
+		self.mute_button.setMouseTracking(True)
+		self.mute_button.installEventFilter(self)
+		self.scene().addWidget(self.mute_button)
+		self.mute_button.resize(40, 40)
+		self.mute_button.move(self.width() - self.mute_button.size().width()/2, self.height() - self.mute_button.size().height()/2)
+		self.mute_button.hide()
+
+	def check_media_status(self, media_status:QMediaPlayer.MediaStatus) -> None:
+		logging.info(media_status)
+		if media_status == QMediaPlayer.MediaStatus.EndOfMedia:
+			#Switches current video to with video at filepath. Reason for this is because Pyside6 is dumb and can't handle setting a new source and freezes.
+			#Even if I stop the player before switching, it breaks.
+			#On the plus side, this adds infinite looping for the video preview.
+			self.player.setSource(QUrl().fromLocalFile(self.filepath))
+			logging.info(f'Set source to {self.filepath}.')
+			# self.video_preview.setSize(self.resolution)
+			
+			self.player.setPosition(0)
+			logging.info(f'Set muted to true.')
+			self.player.play()
+			logging.info(f'Successfully played.')
+			self.keep_controls_in_place()
+			self.update_controls()
+	def update_controls(self) -> None:
+		if self.player.audioOutput().isMuted():
+			self.mute_button.load('./tagstudio/resources/volume_muted.svg')
+		else:
+			self.mute_button.load('./tagstudio/resources/volume_unmuted.svg')
+
+		if self.player.isPlaying():
+			self.play_pause.load('./tagstudio/resources/pause.svg')
+		else:
+			self.play_pause.load('./tagstudio/resources/play.svg')
+	def eventFilter(self, obj:QObject, event:QEvent) -> bool:
+		#This chunk of code is for the video controls.
+		if obj == self.play_pause and event.type() == QEvent.Type.MouseButtonPress and event.button() == Qt.MouseButton.LeftButton:
+			if self.player.hasVideo():
+				self.pause_toggle()
+		
+		if obj == self.mute_button and event.type() == QEvent.Type.MouseButtonPress and event.button() == Qt.MouseButton.LeftButton:
+			if self.player.hasAudio():
+				self.mute_toggle()
+
+		if obj == self.video_preview and event.type() == QEvent.Type.GraphicsSceneHoverEnter or event.type() == QEvent.Type.HoverEnter:
+			if self.video_preview.isUnderMouse():
+				self.underMouse()
+				self.hover_fix_timer.start(10)
+		elif obj == self.video_preview and event.type() == QEvent.Type.GraphicsSceneHoverLeave or event.type() == QEvent.Type.HoverLeave:
+			if not self.video_preview.isUnderMouse():
+				self.hover_fix_timer.stop()
+				self.releaseMouse()
+
+
+		return super().eventFilter(obj, event)
+	
+	def checkIfStillHovered(self) -> None:
```

Comment:
please use `snake_case` instead
```suggestion
	def check_if_still_hovered(self) -> None:
```